src/pydocstyle/parser.py

```diff
@@ -442,6 +442,8 @@ def parse_all(self):
         assert self.current.value == '__all__'
         self.consume(tk.NAME)
         if self.current.value != '=':
+            if self.all is not None:
```

Comment:
A non-literal `__all__` didn't work before, and clearing it would lead to incorrect behaviour in many of my modules unless we could also distinguish between reference and mutation (which is out of scope here!).

Reply:
I think the best way to approach this is to add more unit tests so we can see different cases and how they're handled, for example:

```.py
__all__ = ['a']
__all__ = ['b']
```

```.py
__all__ = ['a']
__all__.append('b')
```

I think it would be best to ignore `__all__` in both these cases, not just go with the first definition. I would say that the rule should be the pydocstyle will only take your `__all__` definition in consideration if it is referenced exactly once in the file.